Extensions/TileMap/tilemapruntimeobject.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Displays a tilemap (mapeditor.org supported).
+ * @memberof gdjs
+ * @class TileMapRuntimeObject
+ * @extends RuntimeObject
+ */
+gdjs.TileMapRuntimeObject = function(runtimeScene, objectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, objectData);
+  this._frameElapsedTime = 0;
+
+  /** @type {number} */
+  this._opacity = objectData.content.opacity;
+  /** @type {string} */
+  this._tilemapJsonFile = objectData.content.tilemapJsonFile;
+  /** @type {string} */
+  this._tilemapAtlasImage = objectData.content.tilemapAtlasImage;
+  /** @type {string} */
+  this._displayMode = objectData.content.displayMode;
+  /** @type {number} */
+  this._layerIndex = objectData.content.layerIndex;
+  /** @type {number} */
+  this._animationSpeed = objectData.content.animationSpeed;
+  /** @type {number} */
+  this._animationFps = objectData.content.animationFps;
+
+  if (this._renderer)
+    gdjs.TileMapRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else
+    this._renderer = new gdjs.TileMapRuntimeObjectRenderer(this, runtimeScene);
+
+  // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+  this.onCreated();
+};
+
+gdjs.TileMapRuntimeObject.prototype = Object.create(
+  gdjs.RuntimeObject.prototype
+);
+gdjs.registerObject('TileMap::TileMap', gdjs.TileMapRuntimeObject);
+
+gdjs.TileMapRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+gdjs.TileMapRuntimeObject.prototype.update = function(runtimeScene) {
+  if (this._animationSpeed <= 0) return;
+  var elapsedTime = this.getElapsedTime(runtimeScene) / 1000; 
+  
+  this._frameElapsedTime += elapsedTime * this._animationSpeed; 
+  // 0.25 = 4 fps , 0.0625 = 1  fps
```

Comment:
Fps means "frame per second" - it's not linear :)
 
1 fps = 1 frame per second => 1/1 second between each frame = 1 seconds
2 fps = 2 frames per second => 1/2 second between each frame = 0.5 seconds
4 fps = 4 frames per second => 1/4 second between each frame = 0.25 seconds
30 fps = 30 frames per second => 1/30 second between each frame = 0.033 seconds
60 fps = 60 frames per second => 1/60 second between each frame = 0.016 seconds

General formula is `fps = 1 / elapsed time (in seconds)` <=> `elapsed time (in seconds) = 1 / fps`

Reply:
updated it now, hope my formula is right now

Reply:
I recommend you read a bit about fps: a fps is a number frame per second. You have to use a division to convert between a time between frame and a frame per second (hence the name per second).
It's not super complex, it's just a matter of using a divison to get the proper quantity:

`fps = 1 / elapsed time (in seconds)` 
and the contrary is:
`elapsed time (in seconds) = 1 / fps`

It's a bit like speed, time and distance : `speed = distance/time` ;) 
Fps is `speed` (much like a mph or a kmh, except that we're talking about a frame per second). Elapsed time is the `time`. And the `distance` is... 1 frame :)

Reply:
this is what it looks like atm:
```
gdjs.TileMapRuntimeObject.prototype.update = function(runtimeScene) {
  if (this._animationSpeedScale <= 0 || this._animationFps === 0) return;
  var elapsedTime = this.getElapsedTime(runtimeScene) / 1000; 
  
  this._frameElapsedTime += elapsedTime * this._animationSpeedScale; 
  while (this._frameElapsedTime > 1 / this._animationFps) {
    this._renderer.incrementAnimationFrameX();
    this._frameElapsedTime -= 1 / this._animationFps;
  }
  
};
```

are you sure you werent looking at an old commit when reviewing it yesterday?

Reply:
It looks good now, with the proper guard against _animationFps being 0 and the rest. I must have been looking at an old file. All good! :)